tests/testthat/test-get_quantile.R

```diff
@@ -1,7 +1,188 @@
+})
+
+
+
+# Requirement T4 ---------------------------------------------------------------
+
+context("get_pvalue - T4. The function accepts a logical argument to request for the confidence intervals of the quantiles")
+
+testthat::test_that("T4.1 No error when the argument to request confidence intervals is logical",{
+
+  survfit_object <- visR::estimate_KM(adtte, strata = "TRTA")
+  testthat::expect_error(visR::get_quantile(survfit_object, conf.int = TRUE), NA)
+})
+
+testthat::test_that("T4.2 An error when the argument to request confidence intervals is not logical",{
+
+  survfit_object <- visR::estimate_KM(adtte, strata = "TRTA")
+  testthat::expect_error(visR::get_quantile(survfit_object, conf.int = "blah"))
+})
+
+testthat::test_that("T4.3 An error when the confidence intervals are requested, but not estimated in the `survfit` object",{
+
+  survfit_object <- visR::estimate_KM(adtte, strata = "TRTA", conf.int = FALSE)
+  testthat::expect_error(visR::get_quantile(survfit_object))
+})
+
+# Requirement T5 ---------------------------------------------------------------
+
+context("get_pvalue - T5. The function is a wrapper around quantile method for `survfit` objects")
+
+testthat::test_that("T5.1 The get_quantiles provides the same information as get_quantiles",{
+
+  survfit_object <- visR::estimate_KM(adtte, strata = "TRTA")
+  quant_surv <- quantile(survfit_object) 
+  
+  q <- do.call(rbind.data.frame, quant_surv)
+  
+  strata <- as.character(unlist(lapply(quant_surv, rownames)))
+  quantity <- unlist(lapply(strsplit(rownames(q), "\\.", fixed = FALSE), `[[`, 1))
+  
+  final <- data.frame(
+    cbind(strata, quantity, q)
+   ,row.names = NULL
+   ,check.names = FALSE
+  )
+  
+  final <- final[ order( final[, "strata"], final[, "quantity"] ),  ]
+
+  testthat::expect_equal(visR::get_quantile(survfit_object), final)
+})
+
+# Requirement T6 ---------------------------------------------------------------
+
+context("get_pvalue - T6. The function returns a dataframe with the requested information")
+
+testthat::test_that("T6.1 The function returns a dataframe",{
+
+  survfit_object <- visR::estimate_KM(adtte, strata = "TRTA")
+  quant <- visR::get_quantile(survfit_object, conf.int = TRUE)
+  
+  testthat::expect_true(class(quant) == "data.frame")
+})
+
+testthat::test_that("T6.2 The output contains a column with the strata names",{
+
+  survfit_object <- visR::estimate_KM(adtte, strata = "TRTA")
+  quant <- visR::get_quantile(survfit_object, conf.int = TRUE)
+  
+  stratNm <- as.factor(unique(names(survfit_object[["strata"]])))
+  
+  testthat::expect_equal(stratNm, unique(quant[["strata"]]))
+})
+
+testthat::test_that("T6.3 The output contains a column with the quantities",{
+
+  survfit_object <- visR::estimate_KM(adtte, strata = "TRTA")
+  quant <- visR::get_quantile(survfit_object, conf.int = TRUE)
+  
+  quaNm <- as.factor(c("lower", "quantile", "upper"))
+  
+  testthat::expect_equal(quaNm, unique(quant[["quantity"]]))
+})
+
```

Comment:
Here:
-- Failure (Line 8): T6.3 The output contains a column with the quantities -------------------------
`quaNm` not equal to unique(quant[["quantity"]]).
'current' is not a factor

Reply:
![image](https://user-images.githubusercontent.com/47894155/119698288-64573f80-be51-11eb-80fd-73aae55ae5cc.png)

Reply:
Hey Rebecca, what is current referring to in your fail message? Are both objects factors for you?

Reply:
Hi Steven, this is not a factor: quant[["quantity"]]

Reply:
Can you reinstall this branch and try again? I think you might have the old function still in your session. There I am getting the same error.

Reply:
Still there

Reply:
Please try again. I made everything a character :(

Reply:
Now it works.